### PR TITLE
Backsub version update and nf-test

### PR DESCRIPTION
**Pull request by kbestak**
Updated Backsub to version v0.4.1 and switched testing to nf-test.

- [x] This comment contains a description of changes (with reason).
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label` -- depends on input data size.
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements. -- not on BioConda, existing Docker image.
- [x] Ensure that the test works with either Docker / Singularity

### CODE REVIEW

File: modules/nf-core/backsub/main.nf [hunk 1]
```diff
@@ -1,28 +1,37 @@
 process BACKSUB {
     tag "$meta.id"
     label 'process_single'
+    tag "modules"
+    tag "modules_nfcore"
+    tag "backsub"
```
Comment (sateeshperi) [inline modules/nf-core/backsub/main.nf:6]: you don't need tags here...only in the test file

File: modules/nf-core/backsub/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+                    [ id:'test' ],
+                    file(params.test_data['imaging']['background_subtraction']['markers'], checkIfExists: true)
+                ]
+                """
+            }
+        }
+        then {
+            assertAll (
+                { assert process.success },
+                { assert process.out.backsub_tif },
```
Comment (sateeshperi) [inline modules/nf-core/backsub/tests/main.nf.test:31]: doesn't tif produce a consistent md5?

File: modules/nf-core/backsub/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+                { assert snapshot(process.out.markerout).match("markerout") },
+                { assert process.out.versions }
```
Comment (sateeshperi) [inline modules/nf-core/backsub/tests/main.nf.test:33]:
```suggestion
                { assert snapshot(process.out.versions).match("versions") }
```

File: modules/nf-core/backsub/tests/main.nf.test (hunk 2 above)
Comment (kbestak) [reply modules/nf-core/backsub/tests/main.nf.test:31]: The metadata in the tif changes (the uuid specifically).

File: modules/nf-core/backsub/main.nf (hunk 1 above)
Comment (kbestak) [reply modules/nf-core/backsub/main.nf:6]: Thanks!

File: modules/nf-core/backsub/tests/main.nf.test.snap [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    "markerout": {
+        "content": [
+            [
+                [
+                    {
+                        "id": "test"
+                    },
+                    "test.csv:md5,28fddf3b89ed0a9b17b1f8f28e754612"
+                ]
+            ]
```
Comment (sateeshperi) [inline modules/nf-core/backsub/tests/main.nf.test.snap:11]: im not seeing the versions in the snapshot

Comment (kbestak) [reply modules/nf-core/backsub/tests/main.nf.test.snap:11]: Thanks for the catch!

**Review by sateeshperi — APPROVED** [review]

**Pull request merged**